tests/Various.ipf

```diff
@@ -346,6 +346,61 @@ Function TC_MMD_bck_REENTRY([md])
 	PASS()
 End
 
+static Function TC_StringDiff()
+	STRUCT IUTF_StringDiffResult result
+	string str
+
+	// DIFF CHECKING
+
+	// Case 1: text is different until line end
+	TC_StringDiff_Check("abcfg", "abcde", "0:3:3> a b c f g", "0:3:3> a b c d e")
+	TC_StringDiff_Check("foo\nabcfg\nbar", "foo\nabcde\nbar", "1:3:7> a b c f g", "1:3:7> a b c d e")
+
+	// Case 2: one line is shorter than the other
+	TC_StringDiff_Check("abc", "abcde", "0:3:3> a b c", "0:3:3> a b c d e")
+	TC_StringDiff_Check("foo\nabc\nbar", "foo\nabcde\nbar", "1:3:7> a b c <\\n>", "1:3:7> a b c d e <\\n>")
+
+	// Case 3: the line endings are different
+	TC_StringDiff_Check("abc\ndef", "abc\rdef", "0:3:3> a b c <\\n>", "0:3:3> a b c <\\r>")
+	TC_StringDiff_Check("abc\ndef", "abc\r\ndef", "0:3:3> a b c <\\n>", "0:3:3> a b c <\\r> <\\n>")
+	TC_StringDiff_Check("abc\r\ndef", "abc\rdef", "0:3:3> a b c <\\r> <\\n>", "0:3:3> a b c <\\r>")
+
+	// Case 4: one string is larger than the other one
+	TC_StringDiff_Check("text\n", "text\nabc", "1:0:5>", "1:0:5> a b c")
```

Comment:
Add a test that contains an empty string and one against case-sensitivity as mentioned above.

The context trimming test is missing the cases where one side has a shorter context than MAX_STRING_DIFF_CONTEXT, e.g. no context on one side.